src/full/Agda/TypeChecking/Warnings.hs

```diff
@@ -76,7 +76,7 @@ instance MonadWarning TCM where
       w' = tcWarning tcwarn
 
       add w tcwarn tcwarns
-        | onlyOnce w && elem tcwarn tcwarns = tcwarns -- Eq on TCWarning only checks head constructor
+        | onlyOnce w && elem tcwarn tcwarns = tcwarns -- Eq on TCWarning only checks for same warning Doc
```

Comment:
The comment was outdated, maybe the whole logic here is outdated as well?

Our collection of warnings is slightly murky: We use a list (allowing duplicates), but then we also use `List.union` to join warning collections, and that function removes duplicates, but only from the second list...
Why not collecting warnings in a `Set` and order them by their `Range` when printing them?

Reply:
I would check what would be the performance impact of using a `Set`. How do we compare the warnings? Though since we are already doing (some) duplicate checking it shouldn't get much worse I believe. Still, perhaps better left for a separate PR.